gremlin-dotnet/src/Gremlin.Net/Driver/Remote/DriverRemoteConnection.cs

```diff
@@ -38,68 +40,69 @@ public class DriverRemoteConnection : IRemoteConnection, IDisposable
+        /// <param name="loggerFactory">A factory to create loggers. If not provided, then nothing will be logged.</param>
         /// <exception cref="ArgumentNullException">Thrown when client is null.</exception>
-        public DriverRemoteConnection(string host, int port, string traversalSource):this(new GremlinClient(new GremlinServer(host, port)), traversalSource)
+        public DriverRemoteConnection(string host, int port, string traversalSource = "g",
+            ILoggerFactory loggerFactory = null) : this(
+            new GremlinClient(new GremlinServer(host, port), loggerFactory: loggerFactory), traversalSource,
+            logger: loggerFactory != null
+                ? loggerFactory.CreateLogger<DriverRemoteConnection>()
+                : NullLogger<DriverRemoteConnection>.Instance)
         {
         }
 
         /// <summary>
-        ///     Initializes a new <see cref="IRemoteConnection" /> using "g" as the default remote TraversalSource name.
+        ///     Initializes a new <see cref="IRemoteConnection" />.
         /// </summary>
         /// <param name="client">The <see cref="IGremlinClient" /> that will be used for the connection.</param>
-        /// <exception cref="ArgumentNullException">Thrown when client is null.</exception>
-        public DriverRemoteConnection(IGremlinClient client):this(client, "g")
+        /// <param name="traversalSource">The name of the traversal source on the server to bind to.</param>
+        /// <exception cref="ArgumentNullException">Thrown when client or the traversalSource is null.</exception>
+        public DriverRemoteConnection(IGremlinClient client, string traversalSource = "g")
+            : this(client, traversalSource, null)
         {
         }
 
-        /// <summary>
-        ///     Initializes a new <see cref="IRemoteConnection" />.
-        /// </summary>
-        /// <param name="client">The <see cref="IGremlinClient" /> that will be used for the connection.</param>
-        /// <param name="traversalSource">The name of the traversal source on the server to bind to.</param>
-        /// <exception cref="ArgumentNullException">Thrown when client is null.</exception>
-        public DriverRemoteConnection(IGremlinClient client, string traversalSource)
+        private DriverRemoteConnection(IGremlinClient client, string traversalSource, string sessionId = null,
+            ILogger<DriverRemoteConnection> logger = null)
         {
             _client = client ?? throw new ArgumentNullException(nameof(client));
             _traversalSource = traversalSource ?? throw new ArgumentNullException(nameof(traversalSource));
-        }
 
-        private DriverRemoteConnection(IGremlinClient client, string traversalSource, Guid sessionId)
-            : this(client, traversalSource)
-        {
-            _sessionId = sessionId.ToString();
+            if (logger == null)
```

Comment:
Logging is not covered by unit tests

Reply:
I usually don't test logging because it's not really part of the business logic, but I guess for an open-source library, it makes sense to at least have some testing. I just added some tests that at least check that some expected log messages are logged and that the `DriverRemoteConnection` can use the `LoggerFactory` that was provided to the `GremlinClient` constructor.

Do you have any other tests in mind?

Reply:
I think enough added tests.